crates/bitwarden-crypto/src/safe/key_wrap.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{CryptoError, EncString, KeyDecryptable, KeyEncryptable, SymmetricCryptoKey};
+
+/// A wrapped symmetric key is an an [EncString], created where a wrapping key is used to encrypt a
+/// key_to_wrap.
+///
+/// Wrapped keys such as cipher keys, or attachment keys, are used to create a layer of indirection,
+/// so that keys can be shared mor granularly, and so that data can be rotated more easily.
+#[derive(Clone, PartialEq)]
+pub struct WrappedSymmetricKey(EncString);
+
+impl From<EncString> for WrappedSymmetricKey {
+    fn from(enc_string: EncString) -> Self {
+        WrappedSymmetricKey(enc_string)
+    }
+}
+
+impl AsRef<EncString> for WrappedSymmetricKey {
+    fn as_ref(&self) -> &EncString {
+        &self.0
+    }
+}
+
+impl From<WrappedSymmetricKey> for EncString {
+    fn from(wrapped: WrappedSymmetricKey) -> Self {
+        wrapped.0
+    }
+}
```

Comment:
I feel like having these two conversions kind of defeat the point of the facade, as they allow you to extract the inner EncString accidentally quite easily.

It might be better to have some explicit functions for access to the inner value (maybe even crate-internal if possible):
```
    pub fn into_inner(self) -> EncString {
        self.0
    }

    pub fn as_inner(&self) -> &EncString {
        &self.0
    }
```

Reply:
Agreed. I mostly want to make it so that the user has to explicitly think about and choose to convert them, I don't think banning conversion outside of the crate is possible (yet?).